index.js

```diff
@@ -6,6 +6,7 @@ var walk = require('jade-walk');
 module.exports = link;
 function link(ast) {
   assert(ast.type === 'Block', 'The top level element should always be a block');
+  if (!ast.nodes.length) return ast;
   var extendsNode = null;
   if (ast.nodes[0].type === 'Extends') {
```

Comment:
Perhaps we could resolve the bug by just testing `ast.nodes && ast.nodes[0].type === 'Extends'` here, rather than special casing the empty file?